server/src/mysql/builder.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use std::{net::SocketAddr, sync::Arc};
+
+use catalog::manager::Manager as CatalogManager;
+use query_engine::executor::Executor as QueryExecutor;
+use snafu::{Backtrace, OptionExt, ResultExt, Snafu};
+use table_engine::engine::EngineRuntimes;
+
+use crate::{instance::InstanceRef, mysql::handler::MysqlHandler};
+
+pub struct Builder<C, Q> {
+    config: Config,
+    runtimes: Option<Arc<EngineRuntimes>>,
+    instance: Option<InstanceRef<C, Q>>,
+}
+
+#[derive(Debug)]
+pub struct Config {
+    pub ip: String,
+    pub port: u16,
+}
+
+#[derive(Debug, Snafu)]
+pub enum Error {
```

Comment:
These `Error`s have reduced a lot. I'm wondering can we move all of them (those defined under `mysql` sub module) to `mysql/mod.rs` for a more concrete error type? E.g., it only needs to export `mysql::Error as MySQLError`, rather than export builder's error, handler's error etc separately.